### DISCUSSION TITLE
Uncaught TileSet2D request abort exception

### DISCUSSION THREAD

**Discussion by aliaksei135** [visgl/deck.gl#8114] [Q&A]
I have an intermittent exception being thrown from a `TileLayer` within a React 18 TS. I haven't been able to work out what is causing it from my side.  I also haven't been able to replicate it in a CodePen, hence discussion instead of issue. Any pointers as to what might be causing this? Thanks in advance!

Here is an excerpt of my code, but it is fairly cookie cutter and close to the demo:

            <DeckGL
                initialViewState={INITIAL_VIEW_STATE}
                controller={true}
                layers={[
                    new TileLayer({
                        id: 'airspaces-layer',
                        data: 'https://api.tiles.openaip.net/api/data/openaip/{z}/{x}/{y}.png?apiKey=' + OPENAIP_API_KEY,
                        minZoom: 5,
                        tileSize: 256,
                        renderSubLayers: props => {
                            const {
                                bbox: {west, south, east, north}
                            }: any = props.tile;

                            return new BitmapLayer(props, {
                                data: undefined,
                                image: props.data,
                                bounds: [west, south, east, north]
                            });
                        },
                        maxCacheSize: 1000,
                    }),
                ]}
                onViewStateChange={({viewState, interactionState, oldViewState}) => {
                    // TODO: Fix stutter of loading when zooming
                    if (!interactionState.isDragging && !interactionState.isPanning && !interactionState.isZooming) {
                          updateCells(viewState);
                    }
                }}
            >
                <ReactMapGL
                    mapboxAccessToken={MAPBOX_TOKEN}
                    mapStyle="mapbox://styles/mapbox/streets-v9"
                    minPitch={0}
                    maxPitch={85}
                    // mapStyle="mapbox://styles/mapbox/satellite-streets-v12"
                />
            </DeckGL>

   
Here is the exception trace.

      Uncaught (in promise) DOMException: The operation was aborted. 
      abort tile-2d-header.js:195
      _pruneRequests tileset-2d.js:338
      update tileset-2d.js:186
      _updateTileset tile-layer.js:147
      updateState tile-layer.js:107
      _update layer.js:842
      _updateLayer layer-manager.js:260
      _updateSublayersRecursively layer-manager.js:213
      _updateLayers layer-manager.js:174
      setLayers layer-manager.js:143
      updateLayers layer-manager.js:151
      _onRenderFrame deck.js:727
      onRender animation-loop.js:232
      _renderFrame animation-loop.js:322
      redraw animation-loop.js:157
      renderFrame animation-loop.js:256
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      requestAnimationFrame request-animation-frame.js:2
      _requestAnimationFrame animation-loop.js:309
      renderFrame animation-loop.js:257
      2 tile-2d-header.js:195

Versions:
- deck.gl: 8.9.28
- Firefox 117

**Comment by Pessimistress** [comment]
If you are using `React.StrictMode`, in dev it mounts the component twice, which could lead to something like this?
